### PR TITLE
Fix for GCC 14

### PR DESCRIPTION
**Pull request by Spacechild1**
fixes https://github.com/supercollider/supercollider/issues/6435

Please merge ASAP :)

### CODE REVIEW

**Review by JordanHendersonMusic — APPROVED** [review]
Thanks for the speedy work :)

**Review by capital-G — CHANGES_REQUESTED** [review]
Thanks for the quick fix!

In general, I'd prefer to do it "properly" by fixing our dependencies if it's not too much work for now.

* It seems our portmidi dependency hasn't been updated since [9 years](https://github.com/supercollider/supercollider/commits/develop/external_libraries/portmidi) - maybe we should update it as it's still maintained, see https://github.com/PortMidi/portmidi ? while we're at it, maybe we can make it a git submodule instead of copying files around to keep our history/repo a bit cleaner?
* Can you estimate how much effort it would take to fix hidapi? IIUC it is a fork maintained by SC, so we could quickly in-cooperate any PRs.

Additionally, if we decide to go with the fix, we should add gcc14 to our linux build pipeline (maybe replacing gcc7/gcc9?). If you need help with this, I can help.
Maybe @dyfer can also provide some input here, as we recently talked about upgrading the versions used in our CI, see https://github.com/supercollider/supercollider/issues/6430

If there is a consensus that this is urgent and should be merged without fixing upstream dependencies - the proposed changes look good to me.

**Comment by Spacechild1** [comment]
> In general, I'd prefer to do it "properly" by fixing our dependencies if it's not too much work for now.

Sure, that's why I said it's a temporary fix. The issue is urgent because developers who have upgraded to GCC 14, like myself, are currently not able to build SC.

**Review by capital-G — COMMENTED** [review]
OK, in this case let's merge it once the pipeline passes.

**Comment by Spacechild1** [comment]
Thanks :)

File: server/scsynth/CMakeLists.txt [hunk 1]
```diff
@@ -54,7 +54,7 @@ elseif(AUDIOAPI STREQUAL portaudio AND SYSTEM_PORTAUDIO)
       message(FATAL_ERROR "Portaudio selected as audio API, but development files not found")
     endif()
 elseif(AUDIOAPI STREQUAL bela)
-    # The Bela lib will have its own set of libraries and includes
+    # The Bela lib will have its own set of libraries and includes
```
Comment (dyfer) [inline server/scsynth/CMakeLists.txt:57]: If we consider this PR, this change probably shouldn't be here :)

**Pull request merged**